Common/src/geometry/CGeometry.cpp

```diff
@@ -4149,3 +4110,58 @@ void CGeometry::SetNaturalElementColoring()
   /*--- In parallel, set the group size to nElem to protect client code. ---*/
   if (omp_get_max_threads() > 1) elemColorGroupSize = nElem;
 }
+
+void CGeometry::ComputeWallDistance(CConfig **config_container, CGeometry ****geometry_container){
+
+  int nZone = config_container[ZONE_0]->GetnZone();
+  bool allEmpty = true;
+  vector<bool> wallDistanceNeeded(nZone, false);
+
+  for (int iInst = 0; iInst < config_container[ZONE_0]->GetnTimeInstances(); iInst++){
+    for (int iZone = 0; iZone < nZone; iZone++){
+
+      /*--- Check if a zone needs the wall distance and store a boolean ---*/
+
+      ENUM_MAIN_SOLVER kindSolver = static_cast<ENUM_MAIN_SOLVER>(config_container[iZone]->GetKind_Solver());
+      if (kindSolver == RANS ||
+          kindSolver == INC_RANS ||
+          kindSolver == DISC_ADJ_RANS ||
+          kindSolver == DISC_ADJ_INC_RANS ||
+          kindSolver == FEM_LES ||
+          kindSolver == FEM_RANS){
+        wallDistanceNeeded[iZone] = true;
```

Comment:
I hope this logic only appears once :)

Reply:
With this implementation I removed even tell of these statements in the driver ;) but we really need some common structure for all this information soon ...